File: Front/CRM/src/app/Modal/Endereco.ts
export class Endereco {
    readonly cep;
    readonly rua;
    readonly id_Endereco;
    readonly complemento;
    readonly cidade;
    readonly estado;
    readonly bairro;

    
    public get Bairro() : string {
        return this.bairro;
    }
    

    public get CEP() : string {
        return this.cep;
    }
    
    
    public get Estado() : string {
        return this.estado;
    }
    
    
    public get Cidade() : string {
        return this.cidade;
    }
    
    public get IdEndereco() : string {
        return this.id_Endereco
    }

    public get Rua() : string {
        return this.rua;
    }
    
    
    public get Complemento() : string {
        return this.complemento;
    }
    
    
    

    constructor({ idEndereco, cep, rua, complemento, cidade, estado, bairro }:
        { idEndereco: string, cep: string, rua: string, complemento: string, 
        cidade: string, estado: string, bairro: string }) {
        this.cep = cep;
        this.cidade = cidade;
        this.complemento = complemento;
        this.estado = estado;
        this.id_Endereco = idEndereco;
        this.rua = rua;
        this.bairro = bairro;
    }
}

File: CRMAPI/Modal/Endereco.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace CRMAPI.Modal
{
    public class Endereco
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string Rua { get; set; }
        public string Complemento { get; set; }
        public string Bairo { get; set; }
        public string CEP { get; set; }

        public string Cidade { get; set; }
        public string Estado { get; set; }

        public Endereco()
        {

        }

        public Endereco(int id, string rua, string complemento, string bairro, string cep, string cidade, string estado)
        {
            this.Id = id;
            this.CEP = cep;
            this.Rua = rua;
            this.Bairo = bairro;
            this.Complemento = complemento;
            this.Estado = estado;
            this.Cidade = cidade;
        }
    }
}
